deploy/roles/math_lm/tasks/main.yml

```diff
@@ -0,0 +1,18 @@
+---
+- name: Verify mathpass file exists
+  stat: path="{{mathpass_location}}"
+  register: mathpass
+  when: update_math_lm_activation_key
+  tags: math_lm
+
+- name: Copy and rename existing mathpass file
+  command: mv {{mathpass_location}} {{mathpass_location}}_{{date}}
+  when: mathpass.stat.exists and update_math_lm_activation_key
+  tags: math_lm
```

Comment:
No need to do all this, just use the `backup` option for the `template` command, which will copy the file with timestamp automatically: http://docs.ansible.com/ansible/latest/template_module.html